.github/workflows/test.yaml

```diff
@@ -63,10 +95,64 @@ jobs:
           make check
     - name: Test Go
       run: go test github.com/snapcore/snapd/...
-  spread:
+  spread-canary:
     needs: unit-tests
-    runs-on: [self-hosted]
+    runs-on: [self-hosted, linux]
+    strategy:
+      fail-fast: false
+      matrix:
+        system:
+        - ubuntu-16.04-64
+        - ubuntu-core-16-64
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Run spread tests
+      run: spread google:${{ matrix.system }}:tests/...
+      # TODO: move this to a reliable post() action.
+    - name: Discard spread workers
```

Comment:
As on master, this TODO can be replaced with setting `if: always()` on the step.  Same for the other job definitions.